apps/system/js/bluetooth_v2.js

```diff
@@ -435,6 +439,52 @@ Bluetooth.prototype = {
   },
 
   /**
+   * Get adapter from bluetooth through promise interface.
+   *
+   * @public
+   * @return {Promise} Bluetooth Adapter
+   */
+  adapter: function bt__adapter() {
+    return new Promise((resolve, reject) => {
+      if (this._adapter !== null) {
+        resolve(this._adapter);
+      } else {
+        this.debug('No BT adapter retrieved');
+        reject();
+      }
+    });
+  },
+
+  /**
+   * Return device pairing result.
+   *
+   * @public
+   * @param {string} mac target device address
+   * @return {Promise} paired result
+   */
+  pair: function bt__pair(mac) {
+    return new Promise((resolve, reject) => {
+      this._adapter.pair(mac).then(() => {
+        resolve(mac);
```

Comment:
Shall we resolve with the `mac` since we request pairing with the `mac` from the caller?